charts/redpanda/servicemonitor.go

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=servicemonitor.go.tpl
+package redpanda
+
+import (
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+)
+
+func ServiceMonitor(dot *helmette.Dot) *monitoringv1.ServiceMonitor {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if !ptr.Deref(values.Monitoring.Enabled, false) {
+		return nil
+	}
+
+	monitorLabels := helmette.Merge(FullLabels(dot), values.Monitoring.Labels)
+
+	matchLabels := map[string]string{
+		"monitoring.redpanda.com/enabled": "true",
+		"app.kubernetes.io/name":          Name(dot),
+		"app.kubernetes.io/instance":      dot.Release.Name,
+	}
+
+	tlsConfigMap := values.Monitoring.TLSConfig
+	var tslConfig *monitoringv1.TLSConfig = nil
+	if len(tlsConfigMap) > 0 {
+		tslConfig = &monitoringv1.TLSConfig{
+			SafeTLSConfig: monitoringv1.SafeTLSConfig{
+				InsecureSkipVerify: false,
+				// TODO, do we support this?
+				// .Values.monitoring.tlsConfig general enough
+			},
+			CAFile:   valueOrDefault(tlsConfigMap, "caFiles", ""),
```

Comment:
Seems like this logic could be made easier by typing the `tlsConfig` as a `monitoringv1.TLSConfig`. Is there any reason to not do so?

Reply:
that definitely would help, i asked, let me make that change and see what happens.